### PR TITLE
gh-122311: Fix some error messages in pickle

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-122311

### CODE REVIEW

File: Lib/test/pickletester.py [hunk 1]
```diff
@@ -1982,8 +1982,10 @@ def test_picklebuffer_error(self):
         pb = pickle.PickleBuffer(b"foobar")
         for proto in range(0, 5):
             with self.subTest(proto=proto):
-                with self.assertRaises(pickle.PickleError):
+                with self.assertRaises(pickle.PickleError) as cm:
                     self.dumps(pb, proto)
+                self.assertEqual(str(cm.exception),
+                    'PickleBuffer can only be pickled with protocol >= 5')
```
Comment (picnixz) [inline Lib/test/pickletester.py:1988]:
```suggestion
                self.assertEqual(str(cm.exception),
                    			 'PickleBuffer can only be pickled with protocol >= 5')
```

File: Lib/pickle.py [hunk 2]
```diff
@@ -314,16 +314,17 @@ def load_frame(self, frame_size):
 # Tools used for pickling.
 
 def _getattribute(obj, name):
+    top = obj
     for subpath in name.split('.'):
         if subpath == '<locals>':
             raise AttributeError("Can't get local attribute {!r} on {!r}"
-                                 .format(name, obj))
+                                 .format(name, top))
```
Comment (picnixz) [inline Lib/pickle.py:321]: Could we take the chance of using f-strings instead?

Comment (serhiy-storchaka) [reply Lib/pickle.py:321]: Maybe in other PRs. I am going to significantly rewrite this code, so using f-strings will be a minor part.

File: Lib/test/pickletester.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/test/pickletester.py:1988]: The line will be too long.

Comment (picnixz) [reply Lib/test/pickletester.py:1988]: Oh ok :( Maybe a temporary variable then? (otherwise, let's leave it like this)

**Review by picnixz — APPROVED** [review]

**Pull request merged**